src/ParallelAlgorithms/Events/ObserveAtExtremum.hpp

```diff
@@ -0,0 +1,404 @@
+          typename... NonTensorComputeTags, typename ArraySectionIdTag>
+template <typename ComputeTagsList, typename DataBoxType,
+          typename Metavariables, size_t VolumeDim, typename ParallelComponent>
+void ObserveAtExtremum<ObservationValueTag, tmpl::list<ObservableTensorTags...>,
+                       tmpl::list<NonTensorComputeTags...>, ArraySectionIdTag>::
+operator()(const typename ObservationValueTag::type& observation_value,
+           const ObservationBox<ComputeTagsList, DataBoxType>& box,
+           Parallel::GlobalCache<Metavariables>& cache,
+           const ElementId<VolumeDim>& array_index,
+           const ParallelComponent* const /*meta*/) const {
+  // Skip observation on elements that are not part of a section
+  const std::optional<std::string> section_observation_key =
+      observers::get_section_observation_key<ArraySectionIdTag>(box);
+  if (not section_observation_key.has_value()) {
+    return;
+  }
+
+  using tensor_tags = tmpl::list<ObservableTensorTags...>;
+
+  // Vector that will contain the local extremum, and the value
+  // of other tensors at that extremum
+  std::vector<double> data_to_reduce{};
+  // Vector containing a description of the data to be reduced.
+  std::vector<std::string> legend{db::tag_name<ObservationValueTag>()};
+  // Location of the local extremum
+  size_t index_of_extremum = 0;
+  // First, look for local extremum of desired scalar
+  tmpl::for_each<tensor_tags>([this, &box, &data_to_reduce, &legend,
+                               &index_of_extremum](auto tag_v) {
+    using tag = tmpl::type_from<decltype(tag_v)>;
+    const std::string tensor_name = db::tag_name<tag>();
+    if (tensor_name == scalar_name_) {
+      if (UNLIKELY(not has_value(get<tag>(box)))) {
+        ERROR("Cannot observe a norm of '"
+              << tensor_name
+              << "' because it is a std::optional and wasn't able to be "
+                 "computed. This can happen when you try to observe errors "
+                 "without an analytic solution.");
+      }
+      const auto& scalar = value(get<tag>(box));
+      const auto [component_names, components] = scalar.get_vector_of_data();
+      if (components.size() > 1) {
+        ERROR("Extremum should be taken on a scalar, yet we have "
+              << components.size() << " components in tensor " << tensor_name);
+      }
+      for (size_t i = 1; i < components[0].size(); i++) {
+        if ((extremum_type_ == "Max" and
+             (components[0][i] > components[0][index_of_extremum])) or
+            (extremum_type_ == "Min" and
+             (components[0][i] < components[0][index_of_extremum]))) {
+          index_of_extremum = i;
+        }
+      }
+      data_to_reduce.push_back(components[0][index_of_extremum]);
+      if (extremum_type_ == "Max") {
+        legend.push_back("Max of " + scalar_name_);
+      } else {
+        legend.push_back("Min of " + scalar_name_);
```

Comment:
(optional) For consistency with `ObserveNorms` I suggest to name this `Max(scalar_name)` and `Min(scalar_name)`